src/domain/core/src/entities/metadata_chain.rs

```diff
@@ -376,7 +376,7 @@ pub enum AppendError {
     #[error(transparent)]
     RefCASFailed(#[from] RefCASError),
     #[error(transparent)]
-    InvalidBlock(#[from] AppendValidationError),
+    InvalidBlock(#[from] Box<AppendValidationError>),
```

Comment:
This looks a quite strange, as at this level reader cannot tell why all other errors are included by-value but this one is boxed. I think a better fix would be to box the whatever field inside the `AppendValidationError` that makes it so big (I assume its `MetadataEvent`).

Reply:
Indeed, I could put the error one level lower and it would be more practical -- thanks for the great tip!

will be implemented

Reply:
Done